### ISSUE TITLE
Disable error detection in doc tests of cfg-disabled items

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#9224]
## Environment

* **IntelliJ Rust plugin version:** 0.4.177.SNAPSHOT-222-dev
* **Rust toolchain version:** 1.63.0 (4b91a6ea7 2022-08-08) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2022.2 Ultimate Edition (IU-222.3345.118)
* **Operating system:** macOS 12.3.1
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.emulate.terminal

## Problem description
The plugin tries not to perform complex analysis (like unresolved references, type mismatched, etc) in cfg-disabled code.
At the same time, if the corresponding item has doc tests in its doc comment, the plugin analyzes the code of the doctest regardless cfg predicate value. It may lead to unexpected errors

<img width="477" alt="Screenshot 2022-08-22 at 15 33 19" src="https://user-images.githubusercontent.com/2539310/185933693-ddaa2c7c-c0b0-4f66-ad32-c797d1b60338.png">

In particular, In faced with this issue in #8961 where plugin doesn't download dependencies not related to the current target. As a result, the plugin started to produce few false positive errors in doc tests of cfg disabled items

<img width="854" alt="Screenshot 2022-08-22 at 15 36 26" src="https://user-images.githubusercontent.com/2539310/185934311-370f4bb9-d2ce-4ca3-813a-8b0fa3c50495.png">

## Steps to reproduce
`lib.rs`
```rust
///
/// ```
/// use rust_project::bar;
/// 
/// bar(123)
/// ```
#[cfg(feature = "foo")]
pub fn foo() {
    bar(123)
}

///
/// ```
/// use rust_project::bar;
///
/// bar()
/// ```
#[cfg(not(feature = "foo"))]
pub fn foo() {
    bar()
}

pub fn bar(#[cfg(feature = "foo")] x: i32) {}
```

`Cargo.toml`
```toml
[package]
name = "rust-project"
edition = "2018"
version = "0.1.0"

[features]
foo = []
```
